kernel/src/scan/state.rs

```diff
@@ -100,6 +99,29 @@ impl DvInfo {
     }
 }
 
+/// utility function for applying a transform expression to convert data from physical to logical
+/// format
+pub fn transform_to_logical(
+    engine: &dyn Engine,
+    physical_data: Box<dyn EngineData>,
+    physical_schema: &SchemaRef,
+    logical_schema: &Schema,
+    transform: &Option<ExpressionRef>,
+) -> DeltaResult<Box<dyn EngineData>> {
+    if let Some(ref transform) = transform {
+        engine
+            .get_expression_handler()
+            .get_evaluator(
+                physical_schema.clone(),
+                transform.as_ref().clone(), // TODO: Maybe eval should take a ref
+                logical_schema.clone().into(),
+            )
+            .evaluate(physical_data.as_ref())
+    } else {
+        Ok(physical_data)
+    }
+}
+
```

Comment:
doc nit: Line 142/143 below might benefit from listing the new `transform` field

Reply:
yep, nice catch, added